apps/dapp-console-api/src/routes/contracts/ContractsRoute.ts

```diff
@@ -48,6 +52,95 @@ export class ContractsRoute extends Route {
       }
     })
 
+  public readonly createContract = 'createContract' as const
+  public readonly createContractController = this.trpc.procedure
+    .use(isPrivyAuthed(this.trpc))
+    .input(
+      this.z.object({
+        contractAddress: zodEthereumAddress,
+        deploymentTxHash: zodEthereumTransaction,
+        deployerAddress: zodEthereumAddress,
+        chainId: this.z.number(),
```

Comment:
tiny nit: you can probably validate the chainId here instead of later on

```typescript
chainId: this.z.number().refine((chainId) => SUPPORTED_CHAINS.some(({ id }) => id === chainId )

```